Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -98,9 +103,122 @@ public virtual Fill Fill(FillModelParameters parameters)
+                Parameters.ConfigProvider
+                    .GetSubscriptionDataConfigs(asset.Symbol)
+                    .IsExtendedMarketHours()))
+            {
+                return fill;
+            }
+
+            // Get the range of prices in the last bar:
+            var high = 0m;
+            var low = 0m;
+            var open = 0m;
+            var endTimeUtc = DateTime.MinValue;
+
+            var subscribedTypes = GetSubscribedTypes(asset);
+            var slip = asset.SlippageModel.GetSlippageApproximation(asset, order);
+
+            if (subscribedTypes.Contains(typeof(Tick)))
+            {
+                var trade = asset.Cache.GetAll<Tick>().LastOrDefault(x => x.TickType == TickType.Trade && x.Price > 0);
+                if (trade != null)
+                {
+                    high = trade.Price;
+                    low = trade.Price;
+                    open = trade.Price;
+                    endTimeUtc = trade.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            else if (subscribedTypes.Contains(typeof(TradeBar)))
+            {
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+                if (tradeBar != null)
+                {
+                    high = tradeBar.High;
+                    low = tradeBar.Low;
+                    open = tradeBar.Open;
+                    endTimeUtc = tradeBar.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            //Get the range of prices in the last bar:
+
+            // do not fill on stale data
+            if (endTimeUtc <= order.Time) return fill;
+
+            //Check if the limit if touched order was filled:
+            switch (order.Direction)
+            {
+                case OrderDirection.Buy:
+                    //-> 1.2 Buy: If Price below Trigger, Buy:
+                    if (low <= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+
+                        if (low <= order.LimitPrice)
+                        {
+                            fill.Status = OrderStatus.Filled;
+                            fill.FillPrice = open <= order.LimitPrice
+                                ? open + slip // Fill price conditional on open
+                                : Math.Max(GetAskPrice(asset, out endTimeUtc), order.LimitPrice) + slip;
+                            // assume the order completely filled
+                            fill.FillQuantity = order.Quantity;
+                        }
+                    }
+
+                    break;
+
+                case OrderDirection.Sell:
+                    //-> 1.2 Sell: If Price above Trigger, Sell:
+                    if (high >= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+
+                        if (high >= order.LimitPrice)
```

Comment:
Cool! So we use trade information to determine if the order is triggered, first if. This second if, should already be using quote information to determine if the limit order is filled. There should be no `slip`. 
If `bidOpen >= order.LimitPrice` think we can fill at `bidOpen`, else `Math.Max(GetCurrentBidPrice(asset, out endTimeUtc), order.LimitPrice)`. Not `ask`, `ask` is for the `OrderDirection.Buy`.

Some references: https://github.com/QuantConnect/Lean/issues/4567 IB SL https://www1.interactivebrokers.com/en/index.php?f=608 IB LIT https://www1.interactivebrokers.com/en/index.php?f=592

Reply:
This wasn't addressed @aarjaneiro? There should be a failing test case for this, let's add a unit test too 👍

Reply:
Apologies -- I thought I changed open -> bidopen!